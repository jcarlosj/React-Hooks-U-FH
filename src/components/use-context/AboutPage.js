import React, { useContext } from 'react';

/** Context */
import { UserContext } from './UserContext';

/** Functional Component */
export const AboutPage = () => {

    const 
        userContext = useContext( UserContext ),      //  Obtenemos valores del Context
        { user, setUser } = userContext;

    console.log( 'userContext', userContext );

    return (
        <section className="container mt-5">
            <h1>About Page</h1>
            {   user.name && <p>Hola, { user.name }.</p> }
            <hr />
            <pre>{ JSON.stringify( user, null, 4 ) }</pre>
            <button
                className="btn btn-warning"
                onClick={ () => setUser( {} ) }
            >
                Logout
            </button>
        </section>
    )
}