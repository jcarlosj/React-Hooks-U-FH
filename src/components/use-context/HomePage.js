import React, { useContext } from 'react';

/** Context */
import { UserContext } from './UserContext';

/** Functional Component */
export const HomePage = () => {

    const 
        userContext = useContext( UserContext ),      //  Obtenemos valores del Context
        { user } = userContext;

    console.log( 'userContext', userContext );

    return (
        <section className="container mt-5">
            <h1>Home Page</h1>
            <hr />
            <pre>{ JSON.stringify( user, null, 4 ) }</pre>
        </section>
    )
}
