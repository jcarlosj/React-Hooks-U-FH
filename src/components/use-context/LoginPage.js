import React from 'react';

/** Functional Component */
export const LoginPage = () => {
    return (
        <section className="container mt-5">
            <h1>Login Page</h1>
            <hr />
        </section>
    )
}