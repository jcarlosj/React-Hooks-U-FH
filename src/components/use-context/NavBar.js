import React from 'react';
import { Link } from 'react-router-dom';

/** Functional Component */
export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-light bg-light">
            <Link className="navbar-brand" to="/">useContext</Link>
            <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                    <li className="nav-item active">
                        <Link className="nav-link" to="/">Home</Link>
                    </li>
                    <li className="nav-item">
                        <Link className="nav-link" to="/about">About</Link>
                    </li>
                    <li className="nav-item">
                        <Link className="nav-link" to="/login">Login</Link>
                    </li>
                </ul>
            </div>
        </nav>
    )
}
