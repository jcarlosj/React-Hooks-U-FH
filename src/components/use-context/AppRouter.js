import React, { version } from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from 'react-router-dom';

/** Components */
import { NavBar } from './NavBar';
import { AboutPage } from './AboutPage';
import { HomePage } from './HomePage';
import { LoginPage } from './LoginPage';

/** Functional Component */
export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <section className="container">
                    <p className="text-right">React v{ version }</p>
                </section>
                <Switch>
                    <Route exact path="/" component={ HomePage } />
                    <Route exact path="/about" component={ AboutPage } />
                    <Route exact path="/login" component={ LoginPage } />
                    <Redirect to="/" />        {/* 404 */}
                </Switch>
            </div>
        </Router>
    )
}
