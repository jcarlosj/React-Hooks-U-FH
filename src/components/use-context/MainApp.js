import React from 'react';

/** Components */
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

/** Functional Component */
export const MainApp = () => {

    /** Valores que deseamos pasar al Contexto (estará disponible para todo el árbol de nodos que abrace el context) */
    const user = {
        id: new Date().getTime(),
        name: 'Eva Sofia',
        email: 'eva.sofia@correo.co'
    }

    return (
        <UserContext.Provider value={ user }>   
            <AppRouter />
        </UserContext.Provider>
    )
}