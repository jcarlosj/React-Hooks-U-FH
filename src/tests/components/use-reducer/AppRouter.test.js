import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

/** Components */
import { AppRouter } from '../../../components/use-context/AppRouter';

/** Contexts */
import { UserContext } from '../../../components/use-context/UserContext'

describe( '<AppRouter />', () => {

    const 
        user = {
            id: 1,
            name: 'Janeth Eva Sofia',
            correo: 'evaÂ©correo.co'
        },
        wrapper = shallow(                      //  Debe usarse 'mount' para acceder a los hijos del componente y poder ver la estructura del Componente HomePage (No se usa por que La version actual de Enzyme no es compatible con React 17) 
            <UserContext.Provider value={{          //  Provee al Componente del contexto que necesita
                user
            }}>   
                <AppRouter />
            </UserContext.Provider>
        );

    test( 'debe desplegar correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    } );

} );