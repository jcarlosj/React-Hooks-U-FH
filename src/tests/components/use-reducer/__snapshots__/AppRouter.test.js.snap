// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> debe desplegar correctamente 1`] = `<AppRouter />`;
